### PR TITLE
Spectramerger sprint

### PR DESCRIPTION
**Pull request by KyowonJeong**
I added SpectraMerger section with a few examples. Also touched Parameter handling part a bit by providing a simple function to print out the full set of parameters with descriptions.

### CODE REVIEW

File: docs/source/user_guide/index.rst [hunk 1]
```diff
@@ -43,6 +43,7 @@ headings and structure.
    smoothing
    centroiding
    spectrum_normalization
+   spectrum_merging
```
Comment (JeeH-K) [inline docs/source/user_guide/index.rst:46]: any reason that the file name is not "spectra_merger"?

File: docs/source/user_guide/spectrum_merging.rst [hunk 2]
```diff
@@ -0,0 +1,341 @@
...
+Block wise spectra merging
+-----------------------
+Our first example merges MS1 spectra block wise.
+
+.. code-block:: python
+  :linenos:
+
+  spectra = exp.getSpectra()
+
+  spectra_ms1 = [s for s in spectra if s.getMSLevel() == 1]
```
Comment (JeeH-K) [inline docs/source/user_guide/spectrum_merging.rst:52]: maybe add a comment: Collecting only MS1 spectra

File: docs/source/user_guide/spectrum_merging.rst [hunk 3]
```diff
@@ -0,0 +1,341 @@
...
+  print(f'Number of MS1 spectra before merge are {len(spectra_ms1)}')
+
+  # merges blocks of MS1
+  merger = oms.SpectraMerger()
+  merger.mergeSpectraBlockWise(exp)
+
+  spectraMerged = exp.getSpectra()
```
Comment (JeeH-K) [inline docs/source/user_guide/spectrum_merging.rst:59]: I suggest to add a comment: get spectra from the updated (merged) experiment

File: docs/source/user_guide/spectrum_merging.rst [hunk 4]
```diff
@@ -0,0 +1,341 @@
...
+  spectraMerged_ms1 = [s for s in spectraMerged if s.getMSLevel() == 1]
+  print(f'Number of MS1 spectra after merge are {len(spectraMerged_ms1)}')
+  # store merged spectra in the disk
+  oms.MzMLFile().store("blockwiseMerged.mzML", exp)
+
+  fig, axs = plt.subplots(6)
```
Comment (JeeH-K) [inline docs/source/user_guide/spectrum_merging.rst:65]: maybe a comment about why there are six subplots: Setting up subplots for five original spectra and merged spectrum

File: docs/source/user_guide/spectrum_merging.rst [hunk 5]
```diff
@@ -0,0 +1,341 @@
...
+  ['controllerType=0 controllerNumber=1 scan=2084', 'controllerType=0 controllerNumber=1 scan=2107']
+
+We can confirm that scans 2077 and 2099 have been merged. In addition, we had a few more pairs of MS2 spectra that were merged. We also plot the input and merged spectra below.
+
+.. code-block:: python
+  :linenos:
+
+
+  # plot the merged and merging MS2 spectra
+
```
Comment (JeeH-K) [inline docs/source/user_guide/spectrum_merging.rst:255]: Unnecessary blank line

File: docs/source/user_guide/spectrum_merging.rst [hunk 6]
```diff
@@ -0,0 +1,341 @@
...
+  # load MS data and store as MSExperiment object
+  exp = oms.MSExperiment()
+  oms.MzMLFile().load('test.mzML', exp)
+  spectra = exp.getSpectra()
+
+  # number of MS1 spectra before averaging
+  spectra_ms1 = [s for s in spectra if s.getMSLevel() == 1]
+  print(f'Number of MS1 spectra before averaging are {len(spectra_ms1)}')
+
+
```
Comment (JeeH-K) [inline docs/source/user_guide/spectrum_merging.rst:305]: here as well

**Review by JeeH-K — CHANGES_REQUESTED** [review]
Adding different examples that added parameters was really helpful to understand. Looks great to me to merge, but maybe after fixing the lint warning. Most of them are just for the quotation marks.

File: docs/source/user_guide/index.rst (hunk 1 above)
Comment (timosachsenberg) [reply docs/source/user_guide/index.rst:46]: I think it is better to call it spectrum_merging as it describes the general process (like the other menu items).

**Comment by timosachsenberg** [comment]
Thanks! I think besides the adding a few comments it is good to go.

**Comment by KyowonJeong** [comment]
Addressed all comments! Done!

**Comment by timosachsenberg** [comment]
There seems to be some linting errors/warnings:
image
but on first sight they don't seem to be critical.

**Comment by timosachsenberg** [comment]
if you click on details you get the warings:
https://github.com/OpenMS/pyopenms-docs/actions/runs/7458254863/job/20291885024?pr=426

**Comment by KyowonJeong** [comment]
> if you click on details you get the warings: https://github.com/OpenMS/pyopenms-docs/actions/runs/7458254863/job/20291885024?pr=426

OOPs, I will check them in no time!

**Comment by KyowonJeong** [comment]
Bugs fixed!

**Comment by timosachsenberg** [comment]
Thanks 🙏

**Pull request merged**